dwi_ml/data/processing/space/neighbourhood.py

```diff
@@ -1,79 +1,85 @@
-                 If radius is a list of floats, returns a multi-radius
-                 neighborhood, with 6 positions per radius (+ 0,0,0).
-                 Good for RNN, for example.
-        "grid":  Returns a 3D volume similar to the original voxel grid,
-                 around (0,0,0,). Dimension 2*radius x 2*radius x 2*radius.
-                 Good for CNN, for example.
+    radius : number (int or float) or list or np.ndarray.
+        Distance to each neighbor sphere (in voxel space).
 
     Returns
     -------
-    directions : np.ndarray with last dimension = 3 (x,y,z coordinate for each
-                 neighbour).
-                 Coordinates are in voxel-space around the (0,0,0) coordinate.
-                 You will need to interpolate the dwi in each direction around
-                 your point of interest to get your neighbourhood.
+    neighborhood : np.ndarray[float]
+        A list of vectors with last dimension = 3 (x,y,z coordinate for each
+        neighbour per respect to the origin). Hint: You can now interpolate your
+        DWI data in each direction around your point of interest to get your
+        neighbourhood.
     """
-    # toDo: This only works with iso voxels! We might have to change this one day
-    #  with radius_vox_x, radius_vox_y, radius_vox_z. But we need to check how
-    #  to find where the x dimension is.
-    #
-    # toDo: add method "n_axes"
-
-    if method == "6axes":
-        unit_axes = _get_6_unit_axes()
-
-        if type(radius_vox) == float:
-            neighborhood = unit_axes * radius_vox
-            neighborhood = np.concatenate(([[0, 0, 0]], neighborhood))
-        else:
-            neighborhood = [[0, 0, 0]]
-            for r in radius_vox:
-                neighborhood = np.concatenate((unit_axes*r, neighborhood))
-    elif method == "grid":
-        if type(radius_vox) == float:
-            neighborhood = []
-            the_range = range(-radius_vox, radius_vox+1)
-            for x, y, z in itertools.product(the_range, the_range, the_range):
-                neighborhood.extend([[x, y, z]])
-            neighborhood = np.asarray(neighborhood)
-        else:
-            raise ValueError("You can't use a list of floats as radius_vox arg"
-                             "together with the grid method.")
+
+    tmp_axes = np.identity(3)
+    unit_axes = np.concatenate((tmp_axes, -tmp_axes))
+
+    if type(radius) == list or type(radius) == np.ndarray:
+        neighborhood = [[0, 0, 0]]
+        for r in radius:
+            neighborhood = np.concatenate((unit_axes*r, neighborhood))
     else:
-        raise NotImplementedError
+        neighborhood = unit_axes * radius
+        neighborhood = np.concatenate(([[0, 0, 0]], neighborhood))
 
     return neighborhood
 
 
-def _get_6_unit_axes():
-    tmp_axes = np.identity(3)
-    unit_axes = np.concatenate((tmp_axes, -tmp_axes))
+# checked!
+def get_neighborhood_vectors_grid(radius: int):
```

Comment:
I think I'll add a `get_neighborhood_vectors_sphere` to get normalized vectors at some point to support my use case, but it's not necessary for now.

Reply:
It can easily be added in another PR.